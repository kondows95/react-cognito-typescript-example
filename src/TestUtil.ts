import {ReactElement} from 'react';
import { render, fireEvent, RenderResult } from '@testing-library/react';
export default class TestUtil
{
  public render: RenderResult;
  constructor(component:ReactElement) {
    this.render = render(component);
  }

  setValue(testId: string, value: string | number) {
    const elem = this.render.getByTestId(testId);
    fireEvent.change(elem, {target: {value: value}})
  }

  click(testId: string) {
    const elem = this.render.getByTestId(testId);
    fireEvent.click(elem);
  }

  getAll(testId: string) {
    return this.render.getAllByTestId(testId);
  }

  get(testId: string) {
    return this.render.getByTestId(testId);
  }
}