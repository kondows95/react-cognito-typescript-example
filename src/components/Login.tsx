import React from 'react';
import Box from '@material-ui/core/Box';
import { withRouter, RouteComponentProps } from 'react-router-dom'
import SignIn from '../containers/auth/SignIn';
import ForgotPassword from '../containers/auth/ForgotPassword';
import ForgotPasswordSubmit from '../containers/auth/ForgotPasswordSubmit';

interface Props extends RouteComponentProps<{}> {
  authState: string | null
}
const Login: React.FC<Props> = ({authState}) => {
   const contents = (
    <React.Fragment>
      <SignIn />
      <ForgotPassword />
      <ForgotPasswordSubmit />
    </React.Fragment>
  );
  return (
    <Box flexGrow={1} textAlign="center" >
      { authState ? contents : null }
    </Box>  
  );
};
export default withRouter(Login);