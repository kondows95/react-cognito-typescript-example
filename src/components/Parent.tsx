import React, {PropsWithChildren} from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../store';

const Parent: React.FC<PropsWithChildren<{}>> = (props) => {
  return (
    <Provider store={store}>
        <Router>
          {props.children}
        </Router>
    </Provider>
  );
};
export default Parent;