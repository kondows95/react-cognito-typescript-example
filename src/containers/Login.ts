import { connect } from 'react-redux';
import Login from '../components/Login';
import { changeAuthState, signOut } from '../modules/auth';
import { AppState, AppDispatch } from '../modules/index';

export const mapStateToProps = (state: AppState) => ({
  authState: state.auth.authState,
});

export const mapDispatchToProps = (dispatch: AppDispatch) => ({
  signOut: () => dispatch(signOut()),
  changeAuthState: (value: string) => dispatch(changeAuthState(value)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);

