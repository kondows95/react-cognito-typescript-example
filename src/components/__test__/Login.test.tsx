/*global jest,expect*/
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import Parent from '../Parent';
import Login from '../Login';
import TestUtil from '../../TestUtil';
let util: TestUtil;

describe("AsuthState is null", () => { 
  beforeEach(() => {
    util= new TestUtil(
      <Parent> 
        <Login authState={null}/> 
      </Parent> 
    );
  });

  it('Snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });
  
  it('Display nothing', () => {
    expect(util.render.container.textContent).toBe("");
  });
});

describe("AsuthState is SignIn", () => { 
  beforeEach(() => {
    util= new TestUtil(
      <Parent> 
        <Login authState={"signIn"}/> 
      </Parent> 
    );
  });

  it('Snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });

  it('SignIn page is displayed', () => {
    expect(util.get("submitButton")).toHaveTextContent("Sign In");
  });
});
