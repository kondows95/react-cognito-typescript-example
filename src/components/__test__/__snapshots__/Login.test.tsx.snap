// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsuthState is SignIn Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <main
      class="MuiContainer-root MuiContainer-maxWidthXs"
    >
      <form>
        <div
          class="MuiBox-root MuiBox-root-14"
        >
          <div
            class="MuiBox-root MuiBox-root-15"
          >
            <div
              class="MuiBox-root MuiBox-root-16"
            >
              <div
                class="MuiBox-root MuiBox-root-17"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="none"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      opacity=".87"
                    />
                  </g>
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-27"
            >
              Sign In
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-28"
          />
          <div
            class="MuiBox-root MuiBox-root-29"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
              >
                Email Address
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="email"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="email"
                  required=""
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-92 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-93"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-94"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
              >
                Password
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-92 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-93"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-95 relative"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              data-testid="submitButton"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Sign In
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root MuiBox-root-129"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              data-testid="forgotPasswordLink"
              href="#"
            >
              Forgot password?
            </a>
          </div>
        </div>
      </form>
    </main>
  </div>
</DocumentFragment>
`;

exports[`AsuthState is null Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2"
  />
</DocumentFragment>
`;
